lib/rack/multipart/parser.rb

```diff
@@ -254,7 +252,6 @@ def handle_fast_forward
         if consume_boundary
           @state = :MIME_HEAD
         else
-          raise EOFError, "bad content body" if @sbuf.rest_size >= @bufsize
```

Comment:
I'm not really convinced that StringScanner + regular expressions is a good approach to parsing data in general. There are too many ways it could go wrong and it's impossible to verify. Getting stuck is a very real concern.

Looking at the original spec, this seems purely defensive, and I'm assuming the spec introduced at the time is still passing even without this line.

Therefore, on that basis alone, I'd say this is still working as expected.